webapp/components/post_view/post_list.jsx

```diff
@@ -171,14 +175,20 @@ export default class PostList extends React.PureComponent {
     }
 
     componentDidUpdate(prevProps) {
+        const prevPosts = prevProps.posts;
```

Comment:
I don't like the look of this. This is a lot of stuff being done every single time the post list re-renders. Is there any way to only do this when a specific state/props value changes? This also seems like it may trigger from random updates and scroll the user away from what they're looking at.